pandas/_libs/missing.pyx

```diff
@@ -220,6 +220,51 @@ cpdef ndarray[uint8_t] isnaobj(ndarray arr):
     return result.view(np.bool_)
 
 
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cpdef ndarray[uint8_t] isna_string(ndarray arr):
+    """
+    Return boolean mask denoting which elements of a 1-D array are na-like,
+    assuming the array only contains pd.NA or strings (guaranteed by the
+    StringDtype).
+
+    This is a special case version of `isnaobj` above, specialized a subset
+    of possible values.
+
+    Parameters
+    ----------
+    arr : ndarray
+
+    Returns
+    -------
+    result : ndarray (dtype=np.bool_)
+    """
+    cdef:
+        Py_ssize_t i, n = arr.size
+        object val
+        bint is_null
+        ndarray result = np.zeros((<object>arr).shape, dtype=np.uint8)
+        flatiter it = cnp.PyArray_IterNew(arr)
+        flatiter it2 = cnp.PyArray_IterNew(result)
+
+    for i in range(n):
+        # The PyArray_GETITEM and PyArray_ITER_NEXT are faster
```

Comment:
Do you know why this is the case? Seems counter intuitive. Especially for the uint8_t array

Reply:
I just copied this from the existing function above ( I assume @jbrockmendel you did those optimizations?) 

To me it doesn't seem counterintuitive that this is faster, as it is using a lower level API with less overhead (and what otherwise happens under the hood anyway, I suppose). I just don't have any intuition about whether is would be significantly faster warranting the added complexity.

Reply:
Based on a quick experiment: compared to using the ndarray `arr[i]`, it makes a big difference and using the ITER trick is a lot faster, but using a memoryview instead (with normal getitem) is then as fast as the code above. Maybe there was a reason in the past that this code couldn't use a memoryview? (eg for const reasons?)

Reply:
IIRC this usage allowed us to support unknown-ndim, whereas casting to memoryview and using `arr[i]` only works if it is 1D

Reply:
Sounds good. Didn't realize this was copy / paste. Changing that (if even required) can be left to another effort